### PR TITLE
Replace repetitive actions with Action Groups in AdminAddBundleItemsTest

### LINKED ISSUE #34312: [Issue] Replace repetitive actions with Action Groups in AdminAddBundleItemsTest
This issue is automatically created based on existing pull request: magento/magento2#34075: Replace repetitive actions with Action Groups in AdminAddBundleItemsTest

---------

### Fixed Issues (if relevant)
Test is refactored according to the best practices followed by MFTF.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by kate-kyzyma**
### Fixed Issues (if relevant)
Test is refactored according to the best practices followed by MFTF.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#34312: Replace repetitive actions with Action Groups in AdminAddBundleItemsTest

### CODE REVIEW

**Comment by kate-kyzyma** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Functional Tests EE

File: app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml [hunk 1]
```diff
@@ -40,39 +40,47 @@
...
         </actionGroup>
-        <checkOption selector="{{AdminAddProductsToOptionPanel.firstCheckbox}}" stepKey="selectFirstGridRow2"/>
-        <click selector="{{AdminAddProductsToOptionPanel.addSelectedProducts}}" stepKey="clickAddSelectedBundleProducts"/>
-        <!-- Check that Bundle Options initialized with default quantity -->
+
+        <!-- Check that Bundle Options initialized with proper quantity -->
         <grabValueFrom selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity('0', '0')}}" stepKey="grabbedFirstBundleOptionQuantity"/>
         <assertEquals stepKey="assertFirstBundleOptionDefaultQuantity">
-            <expectedResult type="string">1</expectedResult>
+            <expectedResult type="string">50</expectedResult>
```
Comment (Den4ik) [inline app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml:73]: It looks little bit strange that PR that replace mftf test with action groups contain expectation of different value.
Could you please explain why expected result changed from 1 to 50?

File: app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml [hunk 2]
```diff
@@ -40,39 +40,47 @@
...
             <actualResult type="string">$grabbedFirstBundleOptionQuantity</actualResult>
         </assertEquals>
         <grabValueFrom selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity('0', '1')}}" stepKey="grabbedSecondBundleOptionQuantity"/>
         <assertEquals stepKey="assertSecondBundleOptionDefaultQuantity">
-            <expectedResult type="string">1</expectedResult>
+            <expectedResult type="string">50</expectedResult>
```
Comment (Den4ik) [inline app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml:78]: Same here

File: app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml [hunk 3]
```diff
@@ -99,53 +111,62 @@
...
         <grabValueFrom selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity('0', '0')}}" stepKey="grabbedFirstBundleOptionQuantityAfterUserInput"/>
         <assertEquals stepKey="assertFirstBundleOptionDefaultQuantityAfterUserInput">
-            <expectedResult type="string">{{BundleProduct.defaultQuantity}}</expectedResult>
+            <expectedResult type="string">50</expectedResult>
             <actualResult type="string">$grabbedFirstBundleOptionQuantityAfterUserInput</actualResult>
         </assertEquals>
         <grabValueFrom selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity('0', '1')}}" stepKey="grabbedSecondBundleOptionQuantityAfterUserInput"/>
         <assertEquals stepKey="assertSecondBundleOptionDefaultQuantityAfterUserInput">
-            <expectedResult type="string">{{BundleProduct.defaultQuantity}}</expectedResult>
+            <expectedResult type="string">50</expectedResult>
```
Comment (Den4ik) [inline app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml:149]: Same here.
Could you please explain why expected result changed from 10 to 50?

File: app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml [hunk 4]
```diff
@@ -99,53 +111,62 @@
...
+            <argument name="inputType" value="{{BundleProduct.optionInputType1}}"/>
         </actionGroup>
-        <checkOption selector="{{AdminProductFormBundleSection.firstProductOption}}" stepKey="selectNewFirstGridRow2"/>
-        <click selector="{{AdminAddProductsToOptionPanel.addSelectedProducts}}" stepKey="clickAddNewSelectedBundleProducts"/>
+
         <!-- Check that existing Bundle Options do not loose user input quantity values  -->
         <grabValueFrom selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity('0', '0')}}" stepKey="grabbedFirstBundleOptionQuantityAfterUserInput"/>
         <assertEquals stepKey="assertFirstBundleOptionDefaultQuantityAfterUserInput">
-            <expectedResult type="string">{{BundleProduct.defaultQuantity}}</expectedResult>
+            <expectedResult type="string">50</expectedResult>
```
Comment (Den4ik) [inline app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml:144]: Same here.
Could you please explain why expected result changed from 10 to 50?

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @kate-kyzyma 
Thanks for your work.
Please check my comments

File: app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml (hunk 1 above)
Comment (kate-kyzyma) [reply app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml:73]: Hi, @Den4ik!
As you may see I use existing ActionGroup here - AddBundleOptionWithTwoProductsActionGroup. Unfortunately, this ActionGroup doesn't allow us to set specific product qty. We have prefiled value - 50.

Code from ActionGroup - AddBundleOptionWithTwoProductsActionGroup:
`<fillField selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity(x, '0')}}" userInput="50" stepKey="fillQuantity1"/>
        <fillField selector="{{AdminProductFormBundleSection.bundleOptionXProductYQuantity(x, '1')}}" userInput="50" stepKey="fillQuantity2"/>`
        
That's why I changed product qty to 50 in the test. As I may see it is not critical for the test.    
If you think it is bad idea, I will try to find some other actions groups to modify the test code.

Comment (Den4ik) [reply app/code/Magento/Bundle/Test/Mftf/Test/AdminAddBundleItemsTest.xml:73]: @kate-kyzyma Thanks for the answer. No all clear for me and looks good

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9221](https://jira.corp.magento.com/browse/ENGCOM-9221) has been created to process this Pull Request
:eight_spoked_asterisk: @Den4ik, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by Den4ik** [comment]
@sidolov Please assign priority to this PR

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-9221](https://jira.corp.magento.com/browse/ENGCOM-9221) has been created to process this Pull Request

**Pull request merged**